### ISSUE TITLE
Suffering to make 'tars base Korean version'

### ISSUE THREAD

**Issue by yspaik** [flairNLP/flair#2841]
I want to make a Korean base tars model that replaces 'tars-base-v8.pt' based on the Korean dataset and the Korean bert pretrained model on the huggingface. The train loss curve falls well as expected, but the dev loss is diverging rather than converging. The dev score also shows a saturation pattern after the initial increase. 

Can you tell me the code or hyperparameters you trained on 'tars-base-v8.pt'? I referenced the Tutorial to make a 'tars base Korean version', I think there seems to be something I'm missing.
https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md

## datasets
https://huggingface.co/datasets/klue
- from subset - ynat, sts, nli are used (properly applicable to text classification)

## parameters

2022-06-27 23:44:41,830 Parameters:
2022-06-27 23:44:41,830  - learning_rate: "0.000040"
2022-06-27 23:44:41,830  - mini_batch_size: "16"
2022-06-27 23:44:41,830  - patience: "10"
2022-06-27 23:44:41,830  - anneal_factor: "0.5"
2022-06-27 23:44:41,830  - max_epochs: "100"
2022-06-27 23:44:41,830  - shuffle: "True"
2022-06-27 23:44:41,830  - train_with_dev: "False"
2022-06-27 23:44:41,830  - batch_growth_annealing: "False"

### Train Loss

![스크린샷 2022-06-28 오전 11 00 14](https://user-images.githubusercontent.com/15726007/176077330-41fe9bb8-e12a-4a6e-bbb7-9c15ad378ab6.png)

Learning Rate

Initially 4e-5
![스크린샷 2022-06-28 오전 11 01 48](https://user-images.githubusercontent.com/15726007/176077503-8ac72b34-0ccf-47fa-956f-4eb3ddb1a83e.png)

Dev Loss
![스크린샷 2022-06-28 오전 11 05 34](https://user-images.githubusercontent.com/15726007/176077594-f05ef82f-690b-47fd-8214-0dd328d60133.png)

Dev Score
![스크린샷 2022-06-28 오전 11 04 16](https://user-images.githubusercontent.com/15726007/176077551-0152e869-4f3d-40c4-9fce-ac191d295009.png)

## codes
```python
class BaseMetaLearner(MetaLearner):

	def __init__(self, *args, **kwargs):
		super(BaseMetaLearner, self).__init__(*args, **kwargs)
		self._lang = 'en'

	def base_learning(
			self,
			embedding: str = 'klue/bert-base',
			down_sample: float = 1.0,
			sample_missing_splits=False,
			corpus_iteration: int = 3,
	):
		assert not self._tars_model and 0 < down_sample <= 1

		corpora = [
			fetch(KlueYnatDataset, sample_missing_splits=sample_missing_splits),
			fetch(KlueNliDataset, sample_missing_splits=sample_missing_splits),
			fetch(KlueStsDataset, sample_missing_splits=sample_missing_splits),
			# fetch(PawsXDataset, sample_missing_splits=sample_missing_splits),
			# fetch(NaverSentimentMovieCommentsDataset, sample_missing_splits=sample_missing_splits),
			# fetch(KoreanRestaurantReviewsDataset, sample_missing_splits=sample_missing_splits),
		]

		tars = TARSClassifier(
			embeddings=embedding,
		)
		# optimizer_params
		# _params = list(tars.tars_model.named_parameters())
		# no_decay = ['bias', 'LayerNorm.bias', 'LayerNorm.weight']
		# decay = 0.01
		# params = [
		# 	{'params': [p for n, p in _params if not any(nd in n for nd in no_decay)], 'weight_decay': decay},
		# 	{'params': [p for n, p in _params if any(nd in n for nd in no_decay)], 'weight_decay': 0.0}
		# ]

		results = []
		for i in range(1, corpus_iteration + 1):
			for c in corpora:
				if 0 < down_sample < 1.0:
					c = copy(c).downsample(percentage=down_sample)

				logger.info(f" start training for corpus {c.name}, {i} -- iteration")
				# tensorboard log directory
				log_dir = self._output_path / 'tensorboard' / f'{c.name}_{i}'
				log_dir.mkdir(parents=True, exist_ok=True)

				if c.name in tars.list_existing_tasks():
					tars.switch_to_task(c.name)
				else:
					label_dict = c.make_label_dictionary(c.name)
					tars.add_and_switch_to_new_task(
						task_name=c.name,
						label_dictionary=label_dict,
						label_type=c.name,
						multi_label=label_dict.multi_label,
					)

				# initialize the text classifier trainer with corpus
				trainer = ModelTrainer(tars, c)
				result = trainer.train(
					base_path=self._output_path / c.name,  # path to store the model artifacts
					learning_rate=self._learning_rate,  # use very small learning rate
					# optimizer=AdamW(params, lr=self._learning_rate, weight_decay=decay),
					optimizer=Adam,
					mini_batch_size=self._mini_batch_size,  # small mini-batch size since corpus is tiny
					patience=self._patience,
					max_epochs=self._max_epochs,  # terminate after 10 epochs
					train_with_dev=self._train_with_dev,
					use_tensorboard=True,
					tensorboard_log_dir=log_dir,
				)
				results.append(result)

		self._tars_model = tars  # replace with fine tuned model
		logger.info(f'fine tuning completed for corpora:{[c.name for c in corpora]}, results:{results}')
		return results


if __name__ == "__main__":
	meta = BaseMetaLearner(
		model_path=None,  # base learning
		max_epochs=100,
		mini_batch_size=16,
		mini_batch_chunk_size=4,
		learning_rate=4e-5,
		train_with_dev=False,
	)
	result = meta.base_learning(down_sample=0.1, embedding="klue/bert-base")
	path = meta.save_model()
	print(path)

```
